packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -170,6 +170,75 @@ export default class RoutesHandler {
         }
     };
 
+    /**
+     * Handler for retrieving existing controller extension data from the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetControllerExtensionData = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const params = req.params as { controllerName: string };
+            const controllerName = sanitize(params.controllerName);
+            const codeExtFiles = await this.readAllFilesByGlob('/**/changes/*_codeExt.change');
+
+            let controllerExists = false;
+            let controllerPath = '';
+            let controllerPathFromRoot = '';
+
+            const project = this.util.getProject();
+            const sourcePath = project.getSourcePath();
+            const rootPath = project.getRootPath();
+            const projectName = rootPath.split(/[\\/]/).slice(-1)[0];
```

Comment:
Here you can use `project.getName()` 😉 See here https://sap.github.io/ui5-tooling/v3/api/@ui5_project_specifications_Project.html

Reply:
This was not in the MiddlewareUtils interface in ui5.d.ts.
That is why I did not use it, thanks, will change this 👍🏻

Reply:
Fixed in: https://github.com/SAP/open-ux-tools/pull/1408/commits/43a23c11efe5bf1c3afb14575909c52931d8f0d1

Reply:
Thx